PostHog/PostHogContext.swift

```diff
@@ -161,4 +153,124 @@ class PostHogContext {
 
         return properties
     }
+
+    private func registerNotifications() {
+        #if os(iOS) || os(tvOS)
+            #if os(iOS)
+                NotificationCenter.default.addObserver(self,
+                                                       selector: #selector(onOrientationDidChange),
+                                                       name: UIDevice.orientationDidChangeNotification,
+                                                       object: nil)
+            #endif
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(onShouldUpdateScreenSize),
+                                                   name: UIWindow.didBecomeKeyNotification,
+                                                   object: nil)
+        #elseif os(macOS)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(onShouldUpdateScreenSize),
+                                                   name: NSWindow.didBecomeKeyNotification,
+                                                   object: nil)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(onShouldUpdateScreenSize),
+                                                   name: NSWindow.didChangeScreenNotification,
+                                                   object: nil)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(onShouldUpdateScreenSize),
+                                                   name: NSApplication.didBecomeActiveNotification,
+                                                   object: nil)
+        #elseif os(watchOS)
+            if #available(watchOS 7.0, *) {
+                NotificationCenter.default.addObserver(self,
+                                                       selector: #selector(onShouldUpdateScreenSize),
+                                                       name: WKApplication.didBecomeActiveNotification,
+                                                       object: nil)
+            } else {
+                NotificationCenter.default.addObserver(self,
+                                                       selector: #selector(onShouldUpdateScreenSize),
+                                                       name: .init("UIApplicationDidBecomeActiveNotification"),
+                                                       object: nil)
```

Comment:
is that available for watchos? https://developer.apple.com/documentation/uikit/uiapplicationdidbecomeactivenotification says its not

Reply:
Surprised that would trigger but it's not documented and in the code is `UIKIT_EXTERN NSNotificationName const UIApplicationDidBecomeActiveNotification API_UNAVAILABLE(watchos) NS_SWIFT_NONISOLATED;` (API_UNAVAILABLE(watchos).
your call here, I'd just double check it's really fired because I'd be super surprised.